src/psyclone/psyir/backend/sir.py

```diff
@@ -117,6 +117,13 @@ def __init__(self, skip_nodes=False, indent_string="  ",
         # found in the PSyIR. This is required as the SIR declares
         # field names after the computation.
         self._field_names = set()
+        # The _scalar_names variable stores the unique scalar names
+        # found in the PSyIR. The current assumption is that scalars
+        # are temporaries. This is not necessarily correct and this
+        # problem is captured in issue #521. Scalar temporaries can be
```

Comment:
"captured" is a strong word - #521 contains no description :-) Please could you add one to that ticket.

Reply:
Point of order. It's an issue ;-)

Reply:
I've added some words to the issue. Hopefully they will be OK.